core/arch/arm/kernel/thread.c

```diff
@@ -195,37 +195,45 @@ static void init_canaries(void)
 #endif/*CFG_WITH_STACK_CANARIES*/
 }
 
-#define CANARY_DIED(stack, loc, n) \
+#define CANARY_DIED(stack, loc, n, addr) \
 	do { \
-		EMSG_RAW("Dead canary at %s of '%s[%zu]'", #loc, #stack, n); \
+		EMSG_RAW("Dead canary at %s of '%s[%zu]' (%p)", #loc, #stack, \
+			 n, (void *)addr); \
```

Comment:
Could replace `(void *)addr` here with `(void *)GET_START_CANARY((stack), (n))`. This would preserve `thread_check_canaries()` implementation that was a little bit easier to read prior this change.
Yet, not a strong opinion.

Reply:
My bad, missing `git add` on my part. I will fix that. I also agree about calling `GET_START_CANARY` in `EMSG_RAW`. Thanks for the feedback.

Reply:
Right!